geoportailv3/static/js/locationinfo/locationinfodirective.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * @fileoverview This file provides a "location information" directive.
+ */
+
+goog.provide('app.locationinfoDirective');
+
+goog.require('app');
+goog.require('app.GetShorturl');
+
+
+/**
+ * @param {string} appLocationinfoTemplateUrl
+ * @return {angular.Directive}
+ * @ngInject
+ */
+app.locationinfoDirective = function(appLocationinfoTemplateUrl) {
+  return {
+    restrict: 'E',
+    scope: {
+      'map': '=appLocationinfoMap',
+      'infoOpen': '=appLocationinfoOpen'
+    },
+    controller: 'AppLocationinfoController',
+    controllerAs: 'ctrl',
+    bindToController: true,
+    templateUrl: appLocationinfoTemplateUrl
+  };
+};
+
+app.module.directive('appLocationinfo', app.locationinfoDirective);
+
+
+
+/**
+ * @constructor
+ * @param {angular.$http} $http
+ * @param {angular.Scope} $scope
+ * @param {app.GetShorturl} appGetShorturl
+ * @param {string} appLocationinfoTemplateUrl
+ */
+app.LocationinfoController =
+    function($http, $scope, appGetShorturl, appLocationinfoTemplateUrl) {
+
+  var map = this['map'];
+
+  /**
+   * @type {ol.FeatureOverlay}
+   * @private
+   */
+  this.featureOverlay_ = this.createFeatureOverlay_();
+
+  $scope.$watch(goog.bind(function() {
+    return this['infoOpen'];
+  }, this), goog.bind(function(newVal) {
+    if (newVal === false) {
+      var features = this.featureOverlay_.getFeatures();
+      features.clear();
```

Comment:
For performance and robustness reasons I think it'd be best to remove the feature overlay from the map (using `this.featureOverlay_.setMap(null)`.